src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java

```diff
@@ -80,20 +80,15 @@ public FlowExecutionList() {
      */
     @Override
     public Iterator<FlowExecution> iterator() {
-        return new AbstractIterator<FlowExecution>() {
+        return new AbstractIterator<>() {
             final Iterator<FlowExecutionOwner> base = runningTasks.iterator();
 
             @Override
             protected FlowExecution computeNext() {
                 while (base.hasNext()) {
                     FlowExecutionOwner o = base.next();
                     try {
-                        FlowExecution e = o.get();
-                        if (e.isComplete()) {
-                            unregister(o);
-                        } else {
-                            return e;
-                        }
+                        return o.get();
```

Comment:
You need to keep the check for `o.isComplete()` but just `continue` (or do nothing) in that case, to find the next one (if any) which is _not_ complete.